modules/SofaSparseSolver/src/SofaSparseSolver/SparseCholeskySolver.cpp

```diff
@@ -108,10 +110,117 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+    
+    perm = new int[A.n ];
+    iperm = new int[A.n ];
```

Comment:
You need to free this allocated memory otherwise there is a memory leak.

I suggest your to use `sofa::type::vector` instead of a C-array.